tests/unit_test/linux/README.md

```diff
@@ -0,0 +1,143 @@
+```
+$ cd afr_root
+$ cmake -B build -DAFR_ENABLE_UNIT_TESTS=on -DBOARD=linux -DVENDOR=pc -DCOMPILER=linux-gcc -DCMAKE_BUILD_TYPE=Debug
+$ cd build
+$ make
+$ make coverage
+blalksdfj
+```
+## Setting up a new Unit Testing module in AFR
+To Setup a module for Unit Testing, as an example we will follow a walkthrough
+approach for **lwip_secure_sockets** which are located in *libraries/abstractions/secure_sockets*. <br>
+
+1. Open the file *tests/unit_test/linux/CMakeLists.txt* for editing.<br>
+    Make sur at around line 15 the the Unit Test directory location exists in this case **../../../libraries** if not add it<br>
+
+``` cmake
+   add_subdirectory(../../../libraries libraries)
+```
+This will make the build system include your files in the build process
+
+2.  add an additional line int the CMakeLists.txt that is located in the directory mentioned above *libraries*
+```cmake
+if (AFR_ENABLE_UNIT_TESTS)
+    add_subdirectory(folder_name/project_name)
+    return()
+endif()
+```
+3. In the directory of the module to be tested create a new directory called *utest*
+
+4. For simplicity`copy the CMakeLists.txt file from *libraries/abstractions/secure_sockets/utest/* into the new Unit Test Directory *utest*
+
+5. Create a test soure file that ends with  *_utest.c* it is where your test code will live<br>
+    you will endup with 2 files in the utest directory (project_name_utest.c and CMakeLists.txt)
+
+6. Edit the copied file (CMakeLists.txt) in the following way
+    1. Modify the *project_name* at the top to match the new one to be tested
+    2. Replace all headers in *mock_list* with the ones you the source file
+       under test uses.
+       ``` cmake
+        list(APPEND mock_list
+                    my_file.h
+                )
+       ```
+    3. Replace all the include directories in *mock_include_list* with the
+       directories your mocks are using
+       ```cmake
+        list(APPEND mock_include_list
+                    /my/file/dir/needs
+                )
+        ```
+    4. Replace the definitions in *mock_define_list* list with the definitions you see fit for your
+       source files
+       ``` cmake
+       list(APPEND mock_define_list
+                    -DMY_DEFINE=1
+                )
+       ```
+       This will create header files in the following form:
+       my_file_mock.h which you will have to include from
+       your test code (project_name_utest.c)
+       This compilation step would create a shared object with the name *project_name_mock.so*
+
+    5. Replace all the files in *real_source_files* with the source files you will
+       be testing. This will create a library with the name *project_file_real.a*
+       ``` cmake
+       list(APPEND real_source_files
+                    code_to_test.c
+                )
+       ```
+    6. Replace all the dirctories in *real_include_directories*  with the
+       directories the source under test will inluce headers from
```

Comment:
typo on include

Reply:
something is wrong with this PR, i guess 2 PR's were merged somehow, i will fix it just to push the readme 
and fix the typos